ddl/reorg.go

```diff
@@ -809,6 +788,27 @@ func (r *reorgHandler) RemoveDDLReorgHandle(job *model.Job, elements []*meta.Ele
 	return removeDDLReorgHandle(r.s, job, elements)
 }
 
+// CleanupDDLReorgHandles removes the job reorganization related handles.
+func CleanupDDLReorgHandles(job *model.Job, pool *sessionPool) {
+	if job != nil && !job.IsFinished() && !job.IsSynced() {
+		// Job is given, but it is neither finished nor synced; do nothing
+		return
+	}
+	sctx, err := pool.get()
+	if err != nil {
+		logutil.BgLogger().Info("CleanupDDLReorgHandles get sessionctx failed", zap.Error(err))
+		return
+	}
+	defer pool.put(sctx)
```

Comment:
Could we use `w.sess.txn()`? I see we have a Txn in `HandleJobDone`(https://github.com/pingcap/tidb/pull/38738/files#diff-dfc42c5764e7e4ff9122a1db728ff1cb0dee56e72dbccefdb211018ccd444c73R698).
The current situation is a bit like this:
```
begin txn1
begin txn2
commit txn2
commit txn1
```
And `cleanDDLReorgHandles` also add a new Txn.

Reply:
Since the code have evolved a bit during my investigation and it is now only called from HandleJobDone, you are correct that we can now simply use `w.sess`, since it was just committed before the call of CleanupDDLReorgHandles.
Let me update the patch like that.